buttercup/cogs/helpers.py

```diff
@@ -0,0 +1,35 @@
+import re
+from datetime import datetime
+
+from discord import DiscordException
+
+
+class NoUsernameError(DiscordException):
+    """Exception raised when the username was not provided."""
+
+    pass
+
+
+def extract_username(display_name: str) -> str:
+    """Extract the Reddit username from the display name."""
+    pattern = re.compile(r"^/?u/(?P<username>\S+)")

```

Comment:
IIRC you'd better define these patterns in separate methods, since they only need to be compiled once.

Reply:
I moved it to a top level definition, is that ok?

Would moving them to separate methods work as well? I'm not sure how exactly Python evaluates stuff.

Reply:
Every time the re.compile method is run, the pattern is compiled. Therefore, placing that function call in a separate function and using this would therefore have the same errors. Making it a top-level definition should definitely work!